compiler/src/main/java/dagger/internal/codegen/BindingGraph.java

```diff
@@ -950,7 +950,8 @@ void resolve(BindingKey bindingKey) {
          * 2. If there are any explicit bindings in this component, they may conflict with those in
          *    the supercomponent, so resolve them here so that conflicts can be caught.
          */
-        if (getPreviouslyResolvedBindings(bindingKey).isPresent()) {
+        if (getPreviouslyResolvedBindings(bindingKey).isPresent()
+            && bindingKey.kind().equals(BindingKey.Kind.CONTRIBUTION)) {
```

Comment:
Hmm. LocalDependencyChecker should still return true if the members injector's transitive dependencies include locally conflicting bindings.

Aha! The issue may be that we're looking for locally-overridden multibindings and optional bindings through the transitive dependencies, but we're looking for locally-conflicting explicit bindings only for the current binding key.

We should look for locally-conflicting explicit bindings in the transitive dependencies, and resolve locally if we find any.  We should be careful because fixing this might uncover existing bugs where some conflicting bindings in subcomponents are currently ignored, but after we fix this they'll be reported.

I think the correct fix is to move the getLocalExplicitBindings call into the LocalDependencyChecker.

Our tests for subcomponent binding conflicts all look at conflicts at the entry point. I bet if we add a test with one nonconflicting binding for the entry points but conflicting bindings for its dependency, it won't fail right now.